### ISSUE TITLE
[Slider] Xml valueFrom and valueTo order problem 

### ISSUE THREAD

**Issue by lperisicz** [material-components/material-components-android#923]
**Description:** When using Slider like this in layout:
<com.google.android.material.slider.Slider
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            android:valueTo="10"
            android:valueFrom="5" /> 
everything works well, but when I switch places of  android:valueTo="10" and  android:valueFrom="5"
while coding or using built in studio code formatter, app crashes with error 

java.lang.IllegalArgumentException: valueFrom must be smaller than valueTo
        at com.google.android.material.slider.Slider.validateValueFrom(Slider.java:397)
        at com.google.android.material.slider.Slider.setValueFrom(Slider.java:441)

**Expected behavior:** I thing order of those two attributes shouldn't matter

**Source code:** 
<com.google.android.material.slider.Slider
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            android:valueFrom="5" 
            android:valueTo="10"/> 

**Android API version:** 28

**Material Library version:** 1.2.0-alpha03

**Device:** Pocophone f1

To help us triage faster, please check to make sure you are using the [latest version](https://github.com/material-components/material-components-android/releases) of the library.

We also happily accept [pull requests](https://github.com/material-components/material-components-android/pulls).

**Comment by RubenGonzalezGonzalez** [comment]
I have the same problem doing programmatically.

**Comment by LinggaMG** [comment]
```
android:valueFrom="5f"
android:valueTo="10f"
```

f => float

**Comment by cketcham** [comment]
Is the issue caused by setting `valueTo` and `valueFrom` in xml? Or just when you call `setValueFrom()` or `setValueTo()`? I could see that you might have an issue calling those methods depending on the order, but I don't see how setting the values in a different order in xml will make a difference since it depends on the order that the values are read in.

Maybe we should have an additional method `setValueRange(float, float)` that lets you set both at the same time so you don't run into that exception?

**Comment by PaulWoitaschek** [comment]
I strongly need a setter for both of the values at once. For my use case the slider represents a playing track so both values can change.
If now the duration of a new track is shorter than the position of the previous track, it throws an exception. If the position of a new track is larger than the duration of the previous track, it throws an exception.
Therefore currently I always need to perform 3 calls.

```kotlin
fun Slider.setRange(from: Float, to: Float) {
  valueFrom = 0F
  valueTo = to
  valueFrom = from
}
```

**Comment by cketcham** [comment]
valueFrom, valueTo, stepSize, and the value(s) of the slider all need to match. For instance, you can't change the stepSize to something else if you already have a value set that doesn't align to the given stepSize.

I'm not sure if it's overkill.. but maybe we need a mechanism to be able to set any number of those things at the same time? That makes me think we need some kind of builder pattern to set the current state, where it will allow setting some subset of the values, and then will perform the validation to make sure it's consistent when set on the slider. How does that sound?

**Comment by drmrbrewer** [comment]
I'm running into this problem, and it's frustrating.  I'm definitely ensuring that `valueTo` is greater than `valueFrom` but it's still giving me the error `valueFrom must be smaller than valueTo`.  I was finding that setting `To` before `From` fixed it, but then had the exception again when trying to set `from` to `-1.0` and `to` to `0.0`.  Surely it shouldn't be this non-intuitive?

**Comment by RudolfHladik** [comment]
this also impact stepSize parameter, when I format the code the order is always `<com.google.android.material.slider.Slider
                    android:layout_width="0dp"
                    android:layout_height="wrap_content"
                    android:layout_marginStart="@dimen/d2"
                    android:layout_weight="1"
                    android:stepSize="@{viewState.enteringDelayModel.step}"
                    android:value="@{viewState.enteringDelayValue}"
                    android:valueFrom="@{viewState.enteringDelayFrom"
                    android:valueTo="@{viewState.enteringDelayTo}"/>`
if I pass with live data step e.g. 5 it calls the setter and validate against yet not set from and to and then it crashes.
if I reorder parameters to be valueTo then valueFrom then stepSize everything works, but it reformats back to "wrong" order when I format the code style

**Comment by opatry** [comment]
It's really cumbersome to follow an initialization order.
Especially when dealing with `RecyclerView` and reusing recycled `Slide` instances.
You should postpone such checks IMO or provide the proper API allow to set all at once.

In the meantime I'm doing:
```
slider.valueTo = Float.MAX_VALUE
slider.valueFrom = Float.MIN_VALUE

slider.valueTo = myMaxValue // hopefully greater than valueFrom…
slider.valueFrom = myMinvalue // hopefully lower than valueTo…
```
Which is really weird…

**Comment by PaulWoitaschek** [comment]
@opatry it's already implemented

**Comment by opatry** [comment]
@PaulWoitaschek so I assume it's not available in `1.2.0-alpha05`, right? When it is supposed to be available publicly?

**Comment by dmgcoding** [comment]
I used seeker instead. It worked perfectly and smoothly for my audio player app.
